geode-management/src/main/java/org/apache/geode/management/api/ClusterManagementServiceTransport.java

```diff
@@ -0,0 +1,104 @@
+
+/**
+ * Interface which abstracts the transport between the CMS client and the endpoint. Currently only
+ * an http implementation exists. However it does allow for different implementations; perhaps
+ * something that doesn't use {@code RestTemplate}. This interface supports the operations from
+ * {@link ClusterManagementService}.
+ */
+@Experimental
+public interface ClusterManagementServiceTransport {
+
+  /**
+   * Submit a message with a specific command. This supports the
+   * {@link ClusterManagementService#create}
+   * and {@link ClusterManagementService#delete} commands.
+   *
+   * @param configMessage configuration object
+   * @param command the command to use
+   * @param responseType the expected response class
+   * @param <T> configuration object which extends {@link AbstractConfiguration}
+   * @return {@link ClusterManagementRealizationResult}
+   */
+  <T extends AbstractConfiguration<?>> ClusterManagementRealizationResult submitMessage(
+      T configMessage, CommandType command,
+      Class<? extends ClusterManagementRealizationResult> responseType);
+
+  /**
+   * Submit a message with a specific command which returns a list result. This supports the
+   * {@link ClusterManagementService#list(AbstractConfiguration)} command.
+   *
+   * @param configMessage configuration object
+   * @param responseType the expected response class
+   * @param <T> configuration object which extends {@link AbstractConfiguration}
+   * @return {@link ClusterManagementListResult}
+   */
+  <T extends AbstractConfiguration<R>, R extends RuntimeInfo> ClusterManagementListResult<T, R> submitMessageForList(
+      T configMessage, Class<? extends ClusterManagementListResult> responseType);
+
+  /**
+   * Submit a message with a specific command which returns a single result. This supports the
+   * {@link ClusterManagementService#get} command.
+   *
+   * @param configMessage configuration object
+   * @param responseType the expected response class
+   * @param <T> configuration object which extends {@link AbstractConfiguration}
+   * @return {@link ClusterManagementGetResult}
+   */
+  <T extends AbstractConfiguration<R>, R extends RuntimeInfo> ClusterManagementGetResult<T, R> submitMessageForGet(
+      T configMessage, Class<? extends ClusterManagementGetResult> responseType);
+
+  /**
+   * Submit a message for a specific command which returns a list result of operations in progress.
+   * This supports the {@link ClusterManagementService#list(ClusterManagementOperation)} command.
+   *
+   * @param responseType the expected response class
+   * @param <A> operation of type {@link ClusterManagementOperation}
+   * @return {@link ClusterManagementListResult}
+   */
+  <A extends ClusterManagementOperation<V>, V extends OperationResult> ClusterManagementListOperationsResult<V> submitMessageForListOperation(
+      A opType, Class<? extends ClusterManagementListOperationsResult> responseType);
```

Comment:
Is there a strong reason not to use `Class<? extends ClusterManagementListOperationsResult<V>` for `responseType`? If not, please use that.

Reply:
We can't do that because of type erasure. To call that method we would have to do something like "ClusterManagementListOperationsResult<V>.class" which isn't allowed.